src/vs/workbench/browser/parts/log/webEmbedderLog.ts

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { LogLevel, LogLevelToString } from 'vs/platform/log/common/log';
+import { Registry } from 'vs/platform/registry/common/platform';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { Extensions, IOutputChannel, IOutputChannelRegistry, IOutputService } from 'vs/workbench/services/output/common/output';
+
+export class WebEmbedderLog extends Disposable {
+	private _outputService: IOutputService;
+
+	constructor(@IOutputService outputService: IOutputService) {
+		super();
+		this._outputService = outputService;
+	}
+
+	public async log(id: string, level: LogLevel, message: string) {
+		const logger = await this.getEmbedderLogChannel();
+
+		const logMessage = this.getLogMessage(id, level, message);
+		logger.append(logMessage);
+	}
+
+	public override async dispose() {
```

Comment:
@connor4312 what's the proper way to ensure this WebEmbedderLog and its output channel gets disposed?